exporter/elasticsearchexporter/data_stream_router.go

```diff
@@ -38,6 +41,14 @@ func routeWithDefaults(defaultDSType, defaultDSDataset, defaultDSNamespace strin
 				return fmt.Sprintf("%s%s%s", prefix, fIndex, suffix)
 			}
 		}
+
+		const otelSuffix = ".otel"
+		// The naming convension for datastream is expected to be the following
+		// "logs-[dataset].otel-[namespace]"
+		if otel && !strings.HasSuffix(dataset, otelSuffix) {
+			dataset += otelSuffix
+		}
+
 		recordAttr.PutStr(dataStreamDataset, dataset)
```

Comment:
This will actually mutate the data which is fine for ecs mode because record level attribute will overwrite scope and resource attributes in the produced doc in ecs. This is a shortcut I took when creating ecs mode data stream routing, and it needs to be cleaned up because it is almost incorrect for otel mode. Ideally the record, scope, and resource attributes should be intact, and only at the doc serialization stage, data_stream.* will be set at top level. I'd consider this as a non-blocker as otel mode is under development.

Reply:
Right, in otel mode, `data_stream.*` attributes should be serialized at the top level and should not be present in `*attributes.data_stream.*`.

Reply:
I can adjust the code to do that. 
Just to confirm that this would be the expectation: we could have a different dataset value at the top from the one that was provided in the attributes.

Reply:
Let's look at an example. If a log record like this comes in:
```yaml
body: "Hello world"
resource:
  attributes:
    data_stream.type: ignored
    data_stream.dataset: lower-precedence
attributes:
  data_stream.dataset: higher-precedence
  foo: bar
```

It should be serialized into a document that looks like this in OTel mode:

```yaml
data_stream:
  type: logs
  dataset: higher-precedence.otel
  namespace: default
body_text: "Hello world"
attributes:
  foo: bar
```